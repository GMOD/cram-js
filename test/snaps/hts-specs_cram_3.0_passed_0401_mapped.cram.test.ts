import { testFile } from '../testUtil'
testFile('hts-specs/cram/3.0/passed/0401_mapped.cram')
