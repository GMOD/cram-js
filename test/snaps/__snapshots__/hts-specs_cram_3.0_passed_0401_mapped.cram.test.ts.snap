// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can dump the whole hts-specs/cram/3.0/passed/0401_mapped.cram without error 1`] = `
[
  {
    "fileId": "_.cram",
    "magic": "CRAM",
    "majorVersion": 3,
    "minorVersion": 0,
  },
  {
    "containerHeader": {
      "_endPosition": 45,
      "_size": 19,
      "alignmentSpan": 0,
      "crc32": 120239702,
      "landmarks": [
        0,
        165,
      ],
      "length": 256,
      "numBases": 0,
      "numBlocks": 2,
      "numLandmarks": 2,
      "numRecords": 0,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 45,
        "_size": 19,
        "alignmentSpan": 0,
        "crc32": 120239702,
        "landmarks": [
          0,
          165,
        ],
        "length": 256,
        "numBases": 0,
        "numBlocks": 2,
        "numLandmarks": 2,
        "numRecords": 0,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 0,
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_I",
              },
              {
                "tag": "LN",
                "value": "1009800",
              },
              {
                "tag": "M5",
                "value": "8ede36131e0dbf3417807e48f77f3ebd",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "ID",
                "value": "samtools",
              },
              {
                "tag": "PN",
                "value": "samtools",
              },
              {
                "tag": "VN",
                "value": "1.10-74-gd7757f2",
              },
              {
                "tag": "CL",
                "value": "samtools fixmate -O sam _.sam -",
              },
            ],
            "tag": "PG",
          },
        ],
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_I",
              },
              {
                "tag": "LN",
                "value": "1009800",
              },
              {
                "tag": "M5",
                "value": "8ede36131e0dbf3417807e48f77f3ebd",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "ID",
                "value": "samtools",
              },
              {
                "tag": "PN",
                "value": "samtools",
              },
              {
                "tag": "VN",
                "value": "1.10-74-gd7757f2",
              },
              {
                "tag": "CL",
                "value": "samtools fixmate -O sam _.sam -",
              },
            ],
            "tag": "PG",
          },
        ],
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": true,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 17,
          },
          "parametersBytes": 1,
        },
        "BB": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 42,
              },
              "parametersBytes": 1,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 37,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 6,
        },
        "BF": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 15,
          },
          "parametersBytes": 1,
        },
        "CF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              3,
            ],
          },
          "parametersBytes": 4,
        },
        "FC": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              98,
            ],
          },
          "parametersBytes": 4,
        },
        "FN": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "FP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MF": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 21,
          },
          "parametersBytes": 1,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              40,
            ],
          },
          "parametersBytes": 4,
        },
        "NP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "NS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "RL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              100,
            ],
          },
          "parametersBytes": 4,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "TS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {},
      "tagIdsDictionary": [
        [],
      ],
    },
    "containerHeader": {
      "_endPosition": 322,
      "_size": 21,
      "alignmentSpan": 300,
      "crc32": 3834170239,
      "landmarks": [
        190,
      ],
      "length": 734,
      "numBases": 200,
      "numBlocks": 10,
      "numLandmarks": 1,
      "numRecords": 2,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 1000,
    },
    "data": [
      {
        "_endPosition": 322,
        "_size": 21,
        "alignmentSpan": 300,
        "crc32": 3834170239,
        "landmarks": [
          190,
        ],
        "length": 734,
        "numBases": 200,
        "numBlocks": 10,
        "numLandmarks": 1,
        "numRecords": 2,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 1000,
      },
      {
        "features": [
          {
            "alignmentStart": 1000,
            "cramFlags": 3,
            "flags": 99,
            "lengthOnRef": 100,
            "mappingQuality": 40,
            "mate": {
              "alignmentStart": 0,
              "flags": 1,
              "readName": undefined,
              "sequenceId": -1,
            },
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "b",
                "data": "ATTTTTCGGGTTTTTTGAAATGAATATCGTAGCTACAGAAACGGTTGTGCACTCATCTGAAAGTTTGTTTTTCTTGTTTTCTTGCACTTTGTGCAGAATT",
                "pos": 1,
                "refPos": 1000,
              },
            ],
            "readGroupId": -1,
            "readLength": 100,
            "readName": "match",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 518,
          },
          {
            "alignmentStart": 1200,
            "cramFlags": 3,
            "flags": 147,
            "lengthOnRef": 100,
            "mappingQuality": 40,
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "b",
                "data": "TTTTTTTAGAAAAATTATTTTTAAGAATTTTTCATTTTAGGAATATTGTTATTTCAGAAAATAGCTAAATGTGATTTCTGTAATTTTGCCTGCCAAATTC",
                "pos": 1,
                "refPos": 1200,
              },
            ],
            "readGroupId": -1,
            "readLength": 100,
            "readName": "match",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 519,
          },
        ],
        "header": {
          "_endPosition": 558,
          "_size": 41,
          "compressedSize": 37,
          "compressionMethod": "raw",
          "content": Uint8Array [
            0,
            131,
            232,
            129,
            44,
            2,
            0,
            8,
            7,
            11,
            12,
            15,
            17,
            21,
            37,
            42,
            255,
            255,
            255,
            255,
            15,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
            0,
          ],
          "contentId": 0,
          "contentPosition": 517,
          "contentType": "MAPPED_SLICE_HEADER",
          "crc32": 2869343137,
          "parsedContent": {
            "_endPosition": 359,
            "_size": 37,
            "contentIds": [
              11,
              12,
              15,
              17,
              21,
              37,
              42,
            ],
            "md5": [
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
            ],
            "numBlocks": 8,
            "numContentIds": 7,
            "numRecords": 2,
            "recordCounter": 0,
            "refBaseBlockId": -1,
            "refSeqId": 0,
            "refSeqSpan": 300,
            "refSeqStart": 1000,
          },
          "uncompressedSize": 37,
        },
      },
    ],
  },
]
`;
